lib/composite_primary_keys/relation.rb

```diff
@@ -1,5 +1,7 @@
 module ActiveRecord
   class Relation
+    AR_CONNECTION_TRILOGY_ADAPTER = ActiveRecord::ConnectionAdapters::TrilogyAdapter
```

Comment:
That will fail if constant is not defined.

Reply:
Right, this assumes Trilogy adapter is loaded which will not be the case the vast majority of times.